### PR TITLE
Update to Style Dictionary v4

### PR DESCRIPTION
**Pull request by c4rlosviteri**
This PR includes an update to SD v4 and its several configurations.

Migration guideline: https://styledictionary.com/version-4/migration/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,27 +36,39 @@
...
-    "require-json5": "^1.1.0",
-    "rimraf": "^3.0.2",
-    "style-dictionary": "^3.7.1",
-    "tinycolor2": "^1.4.2"
+    "query-string": "^9.1.0",
+    "require-json5": "^1.3.0",
+    "rimraf": "^6.0.1",
+    "standard": "^17.1.0",
+    "style-dictionary": "^4.0.1",
+    "tinycolor2": "^1.6.0"
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply package.json:64]: I see that Style Dictionary itself uses tinycolor2 for all their color manipulations. 

For your reference: https://github.com/amzn/style-dictionary/blob/main/lib/common/transforms.js

File: style-dictionary/actions/include-display-less.mjs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import fs from 'fs-extra'
+import path from 'path'
+import { getDirname } from '../utils.mjs'
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply style-dictionary/actions/include-display-less.mjs:3]: No, module JS doesn't have a global `__dirname`, it's back on v20.11 tho, but not in the same way

https://blog.logrocket.com/alternatives-dirname-node-js-es-modules

File: style-dictionary/build.mjs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import { pxToRem } from './transforms/size/px-to-rem.mjs'
+import { stripPx } from './transforms/size/strip-px.mjs'
+import { stripAllPx } from './transforms/size/strip-all-px.mjs'
+import { stripAllPxJs } from './transforms/size/strip-all-px-js.mjs'
+import { float } from './transforms/size/float.mjs'
+import { alpha } from './transforms/color/alpha.mjs'
+import { colorCssTransitive } from './transforms/color/color-css-transitive.mjs'
+import { hex8AndroidTransitive } from './transforms/color/hex8-android-transitive.mjs'
+import { uiColorTransitive } from './transforms/color/uicolor-transitive.mjs'
+import { spTransitive } from './transforms/size/sp-transitive.mjs'
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply style-dictionary/build.mjs:21]: Yes, for now to support `token.attribute.category`

File: style-dictionary/build.mjs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+import { timeSeconds } from './transforms/value/time-seconds.mjs'
+
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply style-dictionary/build.mjs:23]: You can only pass one transform group (that is only a list of transforms), and we are using the `tokens-studio` one

File: style-dictionary/build.mjs [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+// ==== Include custom formats ====
+import { scssMixin } from './formats/scss-mixin.mjs'
+import { scssMap } from './formats/scss-map.mjs'
+import { less as lessFormat } from './formats/less.mjs'
+import { js as jsFormat } from './formats/js.mjs'
+import { site } from './formats/site.mjs'
+import { figma as figmaFormat } from './formats/figma.mjs'
+import { androidColors, androidDimens, androidFontDimens } from './formats/android.mjs'
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply style-dictionary/build.mjs:31]: From the SD v4 source code, just changing the code a bit to support `token.attributes.category`

https://github.com/amzn/style-dictionary/blob/main/lib/common/formats.js

[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply style-dictionary/build.mjs:31]: Most of this code is gonna be removed when the tokens are updated

File: dist/rei-dot-com/scss/media-queries.scss [hunk 6]
```diff
@@ -0,0 +1,98 @@
+// XS
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply dist/rei-dot-com/scss/media-queries.scss:1]: Updated, this was a problem with async await. It's fixed now.

File: style-dictionary/build.mjs (hunk 5 above)
[the remark this answers is not in the document]
Comment (mhewson) [reply style-dictionary/build.mjs:31]: Are you planning to update the tokens as a follow on of as part of this PR?

**Review by mhewson — APPROVED** [review]

File: dist/rei-dot-com/json/global.json [hunk 7]
```diff
@@ -846,25 +816,24 @@
...
-        "category": "color"
+        "deprecated": false
       }
     },
     {
-      "value": "rgba(12, 11, 8, 0.75)",
-      "alpha": 0.75,
-      "category": "color",
+      "$value": "rgba(12, 11, 8, 0.75)",
+      "alpha": ".75,",
```
Comment (mhewson) [inline dist/rei-dot-com/json/global.json:835]: I dont think the alpha should be a string?

File: dist/docsite/figma/figma.json [hunk 8]
```diff
@@ -3,3173 +3,3998 @@
...
         }
       },
       "chip": {
         "default": {
-          "value": "rgba(12, 11, 8, 0.75)",
+          "$value": "#0c0b08",
+          "$type": "color",
           "alpha": 0.75,
-          "type": "color"
+          "filePath": "tokens/global/chip.json5"
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply dist/docsite/figma/figma.json:91]: Yes, that's the plan

File: dist/docsite/figma/figma.json [hunk 9]
```diff
@@ -3,3173 +3,3998 @@
...
-        "value": "Graphik, \"Graphik fallback\", \"Helvetica Neue\", sans-serif",
+        "$value": "Graphik, \"Graphik fallback\", \"Helvetica Neue\", sans-serif",
         "mixin": "textBody300",
-        "property": "font-family"
+        "property": "font-family",
+        "filePath": "tokens/web/text-body.json5"
       },
       "style": {
-        "value": "normal",
+        "$value": "normal",
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply dist/docsite/figma/figma.json:2149]: It's the new DTCG syntax

**Review by mhewson — APPROVED** [review]

File: dist/docsite/figma/figma.json [hunk 10]
```diff
@@ -3,3173 +3,3998 @@
...
-          "type": "letter-spacing",
+          "$value": "0px",
+          "$type": "dimension",
           "mixin": "textHeadingSerif300",
-          "property": "letter-spacing"
+          "property": "letter-spacing",
+          "filePath": "tokens/web/text-heading.json5",
+          "$extensions": {
+            "studio.tokens": {
+              "originalType": "letterSpacing"
```
[the remark this answers is not in the document]
Comment (c4rlosviteri) [reply dist/docsite/figma/figma.json:2799]: Yes

File: package.json [hunk 11]
```diff
@@ -35,31 +36,43 @@
...
+    "lint": "eslint \"style-dictionary/**/*.mjs\""
   },
   "devDependencies": {
     "@divriots/style-dictionary-to-figma": "^0.4.0",
+    "@tokens-studio/sd-transforms": "^1.2.2",
     "concat": "^1.0.3",
     "cross-env": "^7.0.3",
     "dir-to-json": "^1.0.0",
-    "fs-extra": "^10.1.0",
+    "eslint": "^8.57.0",
```
Comment (sikhote) [inline package.json:54]: Thanks for adding this in!

**Review by sikhote — APPROVED** [review]

**Review by mhewson — APPROVED** [review]

**Pull request merged**